modules/tensor_mechanics/src/materials/ComputeFiniteStrain.C

```diff
@@ -174,3 +128,91 @@ ComputeFiniteStrain::computeQpStrain()
   _mechanical_strain[_qp] = _rotation_increment[_qp] * _mechanical_strain[_qp] * _rotation_increment[_qp].transpose();
   _total_strain[_qp] = _rotation_increment[_qp] * _total_strain[_qp] * _rotation_increment[_qp].transpose();
 }
+
+void
+ComputeFiniteStrain::computeQpIncrements(RankTwoTensor & total_strain_increment, RankTwoTensor & rotation_increment)
+{
+  if (_decomposition_method == "Rashid")
```

Comment:
There's nothing in the MOOSE coding standards that says we should use a switch.  If we really want to enforce that, someone should add it to the standards. I can't imagine that it makes any appreciable performance difference here.

, and I personally find if - then